src/components/Quiz/utils.ts

```diff
@@ -0,0 +1,86 @@
+import { USER_STATS_KEY } from "../../constants"
+
+import { CompletedQuizzes, QuizShareStats } from "../../types"
+
+import allQuizzesData from "../../data/quizzes"
+
+export const getTotalQuizzesPoints = () =>
+  Object.values(allQuizzesData)
+    .map((quiz) => quiz.questions.length)
+    .reduce((accumulator, currentValue) => {
+      return accumulator + currentValue
+    }, 0)
+
+export const getNumberOfCompletedQuizzes = (quizzes: CompletedQuizzes) =>
+  Object.values(quizzes)
+    .map((v) => v[0])
+    .filter((v) => v).length
+
+export const updateUserStats = ({
+  average,
+  completed,
+  numberOfCorrectAnswers,
+  quizKey,
+  quizScore,
+  setUserStats,
+  userScore,
+}) => {
+  const userStats = JSON.parse(localStorage.getItem(USER_STATS_KEY)!)
+  const completedQuizzes = JSON.parse(completed)
+
+  // Get previous score on quiz to compare on retry, if previous score is higher then keep it
+  const lastScore = completedQuizzes[quizKey][1]
+  // Update user score, average and save to local storage
+  const newUserScore = userScore + numberOfCorrectAnswers - lastScore
+  const newAverage = [...average, quizScore]
+  const newCompleted = JSON.stringify({
+    ...completedQuizzes,
+    [quizKey!]: [
+      quizScore === 100,
+      quizScore > lastScore ? numberOfCorrectAnswers : lastScore,
+    ],
+  })
+
+  if (numberOfCorrectAnswers > lastScore) {
+    setUserStats({
+      ...userStats,
+      score: newUserScore,
+      average: newAverage,
+      completed: newCompleted,
+    })
+
+    localStorage.setItem(
```

Comment:
@pettinarip stats should be updated also when a user completes a quiz that's outside of the hub, not only inside the hub page

Reply:
Ofc, happy to pair or jump into a call to talk more about this if you are interested.

Reply:
@pettinarip this would imply rewriting a lot of the current logic too (e.g.: calculate when a quiz is finished is not trivial), which is out of scope the this epic, could be considered for a future iteration